### PR TITLE
ZIP-10: Governance

### PR DESCRIPTION
**Pull request by ansnunez**

### CODE REVIEW

**Comment by bb111189** [comment]
For results, we should be looking at 3 key metrics 
- Has the vote reached a minimum quorum? (eg 30%)
- Has the vote exceeded agreement thresholds (eg. 50%)

File: zips/zip-10.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+| 10  | Governance | Draft | Standards Track  | Antonio Nunez <antonio@zilliqa.com> | 2020-07-24 | 2020-07-24
+
+
+## Abstract
+
+This ZIP introduces the support for governance by leveraging on miners' proof-of-work (PoW) submissions to cast votes on proposals.
+
+## Motivation
+
+The consensus protocol and immutable nature of blockchain provide useful tools for decision-making among multiple participants. In the context of the Zilliqa Mainnet, for instance, miners are those participants, and decision-making can include proposals put forward by Zilliqa Research that require feedback (in the form of votes) from the mining community. Those votes can be tallied and made part of the blocks generated by the Mainnet, providing a permanent record that can be easily referenced by any party.
```
Comment (bb111189) [inline zips/zip-10.md:12]:
```suggestion
The consensus protocol and immutable nature of blockchain provide useful tools for decision-making among multiple participants. In the context of the Zilliqa Mainnet, for instance, miners are those participants, and decision-making can include proposals put forward by Zilliqa Research or Zilliqa community that require feedback (in the form of votes) from the mining community. Those votes can be tallied and made part of the blocks generated by the Mainnet, providing a permanent record that can be easily referenced by any party.
```

File: zips/zip-10.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+### On-chain Changes
+
+Governance can be supported with potentially minimal changes to the core protocol.
+
+1. At the start of the DS epoch, mining nodes add an optional byte to their PoW submission messages. This byte represents the miner's vote, the value of which is specified by the different options listed in the published proposal (e.g., 1=Yes, 2=No).
+1. The DS block header format is modified to include the counts for each vote value received by the DS committee. Since the counts are part of the header, they are naturally included in the DS block consensus performed by the committee.
+
+Some implementation considerations:
+
+1. The mining node's local API server (which listens at `STATUS_RPC_PORT`) needs to support a new API request message to set the vote value for the next DS epoch.
```
Comment (bb111189) [inline zips/zip-10.md:36]:
```suggestion
1. The mining node's local API server (which listens at `STATUS_RPC_PORT`) needs to support a new API request message to set the proposal id and vote value for the next pre-set number of DS epoch.
```

Proposal id will be important so as to ensure, miners do not blindly run a script to vote blindly.

**Comment by bb111189** [comment]
We will also need to think of how DS members can vote on the issue.

**Comment by ansnunez** [comment]
> We will also need to think of how DS members can vote on the issue. A well-performing DS node can effectively stay in DS committee due to the reputation mechanism. As such, they may not be able to do voting via PoW

Good catch. I added another note at the end of Specification section.

**Comment by ansnunez** [comment]
> For results, we should be looking at a few metrics
> 
> * Has the vote reached a minimum quorum? (eg 30%)
> * Has the vote exceeded agreement thresholds (eg. 50%)
> 
> We can parameterize all these values

Added note to the end of Off-chain Components section

**Pull request merged**